src/Explorer/Panes/SettingsPaneR/index.tsx

```diff
@@ -0,0 +1,433 @@
+import { PrimaryButton } from "office-ui-fabric-react";
+import React, { FunctionComponent, KeyboardEvent, useEffect, useLayoutEffect, useRef, useState } from "react";
+import * as Constants from "../../../Common/Constants";
+import { configContext } from "../../../ConfigContext";
+import { LocalStorageUtility, StorageKey } from "../../../Shared/StorageUtility";
+import * as StringUtility from "../../../Shared/StringUtility";
+import * as NotificationConsoleUtils from "../../../Utils/NotificationConsoleUtils";
+import Explorer from "../../Explorer";
+
+export interface SettingsPaneProps {
+  explorer: Explorer;
+  closePanel: () => void;
+}
+
+export const SettingsPaneR: FunctionComponent<SettingsPaneProps> = ({ explorer, closePanel }: SettingsPaneProps) => {
+  const [formErrors, setFormErrors] = useState("");
+  const [formErrorsDetails, setFormErrorsDetails] = useState();
+  const [isExecuting, setIsExecuting] = useState(false);
+  const [pageOption, setPageOption] = useState("");
+  const [customItemPerPage, setCustomItemPerPage] = useState(0);
+  const [crossPartitionQueryEnabled, setCrossPartitionQueryEnabled] = useState(
+    LocalStorageUtility.hasItem(StorageKey.IsCrossPartitionQueryEnabled)
+      ? LocalStorageUtility.getEntryString(StorageKey.IsCrossPartitionQueryEnabled) === "true"
+      : true
+  );
+  const [graphAutoVizDisabled, setGraphAutoVizDisabled] = useState(
+    LocalStorageUtility.hasItem(StorageKey.IsGraphAutoVizDisabled)
+      ? LocalStorageUtility.getEntryString(StorageKey.IsGraphAutoVizDisabled)
+      : "false"
+  );
+  const [maxDegreeOfParallelism, setMaxDegreeOfParallelism] = useState(
+    LocalStorageUtility.hasItem(StorageKey.MaxDegreeOfParellism)
+      ? LocalStorageUtility.getEntryNumber(StorageKey.MaxDegreeOfParellism)
+      : Constants.Queries.DefaultMaxDegreeOfParallelism
+  );
+  const explorerVersion = configContext.gitSha;
+  const shouldShowQueryPageOptions = explorer?.isPreferredApiDocumentDB();
+  const shouldShowGraphAutoVizOption = explorer?.isPreferredApiGraph();
+
+  const shouldShowCrossPartitionOption = !explorer?.isPreferredApiGraph();
+  const shouldShowParallelismOption = !explorer?.isPreferredApiGraph();
+
+  const pageOptionsRef = useRef<HTMLLabelElement>();
+  const displayQueryRef = useRef<HTMLLabelElement>();
+  const maxDegreeRef = useRef<HTMLInputElement>();
+
+  useEffect(() => {
+    _loadSettings();
+  }, []);
+
+  useLayoutEffect(() => {
+    if (explorer?.isPreferredApiGraph()) {
+      displayQueryRef?.current?.focus();
+    } else if (explorer?.isPreferredApiTable()) {
+      maxDegreeRef?.current?.focus();
+    }
+    pageOptionsRef?.current?.focus();
+  }, []);
```

Comment:
What is this for? Can it be removed? I would prefer we use the native focusing behavior of the browser

Reply:
![image](https://user-images.githubusercontent.com/80053762/111181129-a94e3280-85d3-11eb-8557-bd661dec56b5.png)
https://reactjs.org/docs/refs-and-the-dom.html

Reply:
done